dss/util/auth/auth0.py

```diff
@@ -24,21 +24,41 @@ def security_flow(self, *args, **kwargs):
         kwargs of this method, and used to call the correct method.
         """
         #  TODO add some type of jwt inspection
-        requested_method = kwargs.get('auth_method').lower()
-        if requested_method is None or requested_method not in self.valid_methods.keys():
-            err = f'Unable to locate auth_method {requested_method} for request, valid methods are: '
-            err += f'{", ".join(self.vaid_methods)}'
-            raise DSSException(500, err)
+
+        # Get name of method to use
+        if 'auth_method' in kwargs:
+            method = kwargs['auth_method']
+        elif len(args) > 0:
+            method = args[0]
+            kwargs['auth_method'] = method
         else:
-            executed_method = self.valid_methods[requested_method]
-            executed_method(*args, **kwargs)
+            raise RuntimeError("Error: invalid arguments passed to Auth0 security_flow() method")
+
+        # Ensure method is valid
+        if method is None or method not in self.valid_methods.keys():
+            err = f'Unable to locate auth_method {method} for request, valid methods are: '
+            err += f'{", ".join(self.valid_methods)}'
+            raise DSSException(500, err)
+
+        # Any further kwarg processing should happen
+        # from inside the method that needs that info,
+        # to limit specifications.
+
+        # Dispatch to correct method
+        executed_method = self.valid_methods[method]
+        executed_method(*args, **kwargs)
 
     def _create(self, *args, **kwargs):
         """Auth checks for any 'create' API endpoint actions"""
-        if kwargs.get('security_groups') is None:
-            groups = args[0]
-        else:
+        # Get name of allowed groups (either security_groups kwarg or second positional arg)
+        if 'security_groups' in kwargs:
```

Comment:
This block probably looks familiar from the other PR and the one in Fusillade, but not quite again, since for Auth0 the security groups are specified as the second positional arg